adapters/trustedstack/trustedstack.go

```diff
@@ -0,0 +1,138 @@
+package trustedstack
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+
+	reqJson, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    reqJson,
+		Headers: headers,
+		ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+	}}, errs
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var errs []error
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+
+	var bidResp openrtb2.BidResponse
+
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponse()
+
+	for _, sb := range bidResp.SeatBid {
+		for i := range sb.Bid {
+			bidType, err := getMediaTypeForImp(&sb.Bid[i], internalRequest.Imp)
+			if err != nil {
+				errs = append(errs, err)
+			} else {
+				b := &adapters.TypedBid{
+					Bid:     &sb.Bid[i],
+					BidType: bidType,
+				}
+				bidResponse.Bids = append(bidResponse.Bids, b)
+			}
+		}
+	}
+	return bidResponse, errs
+}
+
+// Builder builds a new instance of the Trustedstack adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	url := buildEndpoint(config.Endpoint, server.ExternalUrl)
+	return &adapter{
+		endpoint: url,
+	}, nil
+}
+
+func getMediaTypeForImp(bid *openrtb2.Bid, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	mediaType, err := getBidMediaTypeFromMtype(bid)
+	if err == nil {
+		return mediaType, nil
+	}
+	mediaType = openrtb_ext.BidTypeBanner
+	for _, imp := range imps {
+		if imp.ID == bid.ImpID {
+			switch {
+			case imp.Banner == nil && imp.Video != nil && imp.Audio == nil && imp.Native == nil:
+				mediaType = openrtb_ext.BidTypeVideo
+			case imp.Banner == nil && imp.Video == nil && imp.Audio != nil && imp.Native == nil:
+				mediaType = openrtb_ext.BidTypeAudio
```

Comment:
refer https://github.com/prebid/prebid-server/pull/3618/files#r1570529242

`static/bidder-info/trustedstack.yaml` doesnot define `audio` as supported media type. 

Should make changes in adapter code to only support media types defined in bidder info yaml or update bidder info yaml to define `audio` as supported media type as well.

```
capabilities:
  app:
    mediaTypes:
      - banner
      - video
      - native
  site:
    mediaTypes:
      - banner
      - video
      - native
```

Reply:
> Shall bidder response parsing however can support all the media types and that can be utilised when the capabilities are updated?

@product-trustedstack for now adapter code should only check for media types supported. Code can be updated  when capabilities includes new media type

Reply:
This was the fallback logic as discussed in the comment https://github.com/prebid/prebid-server/pull/3618#discussion_r1570583632.
This has been removed. The core logic to get the mtype from the field above has been updated to only support media types currently mentioned in the config.